pq/pq.go

```diff
@@ -154,25 +174,56 @@ func (q *Queue) Pending() int {
 // Writer returns the queue writer for inserting new events into the queue.
 // A queue has only one single writer instance, which is returned by GetWriter.
 // The writer is is not thread safe.
-func (q *Queue) Writer() *Writer {
-	return q.writer
+func (q *Queue) Writer() (*Writer, error) {
+	if q.writer != nil {
+		return q.writer, nil
+	}
+
+	rootBuf, err := q.accessor.ReadRoot()
+	if err != nil {
+		panic("TODO")
+	}
+
+	root := castQueueRootPage(rootBuf[:])
+	tail := q.accessor.ParsePosition(&root.tail)
+
+	writeBuffer := q.settings.WriteBuffer
+	flushed := q.settings.Flushed
+	writer, err := newWriter(&q.accessor, q.pagePool, writeBuffer, tail, flushed)
+	if err != nil {
+		panic("TODO")
```

Comment:
?

Reply:
oh, nice one. Need to grep for more TODOs :)